create_archive.py

```diff
@@ -5,98 +5,136 @@
+        if t1_file:
+            t1_file = os.path.join(t1_anat_dir, t1_file[0])
+            anat_dir = os.path.join(rootdir, sub_dir, ses_dir, 'anat')
+            if not os.path.isdir(anat_dir):
+                os.mkdir(anat_dir)
+            dest_file = os.path.join(anat_dir, sub_dir + '_' + ses_dir + '_T1w.nii.gz')
+            if os.path.exists(dest_file):
+                print('Found downloaded T1 file - overwriting!')
+                os.remove(dest_file)
+                os.remove(dest_file.replace('.nii.gz', '.json'))
+            shutil.copyfile(t1_file, dest_file)
+
+    t2_anat_dir = os.path.join(flywheel_basedir, 'input', 't2w_anatomy')
+    if os.path.isdir(t2_anat_dir):
+        t2_file = os.listdir(t2_anat_dir)
+        if t2_file:
+            anat_dir = os.path.join(rootdir, sub_dir, ses_dir, 'anat')
+            if not os.path.isdir(anat_dir):
+                os.mkdir(anat_dir)
+            t2_file = os.path.join(t2_anat_dir, t2_file[0])
+            dest_file = os.path.join(anat_dir, sub_dir + '_' + ses_dir + '_T2w.nii.gz')
+            if os.path.exists(dest_file):
+                print('Found downloaded T2 file - overwriting!')
+                os.remove(dest_file)
+                os.remove(dest_file.replace('.nii.gz', '.json'))
+            shutil.copyfile(t2_file, dest_file)
+
+
+if __name__ == '__main__':
+    ### SETUP
+    # Define variables
+    flywheel_basedir = os.environ['FLYWHEEL']
+    rootdir = os.path.join(flywheel_basedir, 'input', 'bids_dataset')
+    if not os.path.exists(rootdir):
+        os.makedirs(rootdir)
+
+    # Read in config file
+    config_file = os.path.join(flywheel_basedir, 'config.json')
+    if not os.path.exists(config_file):
+        raise Exception('Config file (%s) does not exist' % config_file)
+    fp = open(config_file, 'r')
+    config_contents = json.loads(fp.read())
+    fp.close()
+    # Get apikey and session ID number from config file
+    api_key = str(config_contents['inputs']['api_key']['key'])
+    analysis_id = str(config_contents['destination']['id'])
+
+    ## Create SDK client
+    print("Create SDK client")
+    fw = flywheel.Flywheel(api_key)
+
+    # Get analysis
+    analysis = fw.get_analysis(analysis_id)
+    # Get container type and id from
+    container_type = analysis['parent']['type']
+    container_id = analysis['parent']['id']
+
+    # Check if curate-bids was ran
+    # Determine if ID is a project or a session ID
+    if container_type == 'project':
+        # Get list of sessions within project
+        container = fw.get_project(container_id)
+    elif container_type == 'session':
+        # If container type is a session, get the specific session
+        container = fw.get_session(container_id)
+
+    BIDS_metadata = container.get('info', {}).get('BIDS')
+    if BIDS_metadata:
+        try:
+            export_bids.export_bids(fw, rootdir, None, container_type=container_type, container_id=container_id)
+            if BIDS_metadata != 'NA':
```

Comment:
So thinking on it, I'm pretty sure export bids would do nothing in the case that the session BIDS object is 'NA', but it would not error, so I added a log statement. Although it may be a better idea to go ahead and treat it as though export did fail in that case.